dateutil/parser/isoparser.py

```diff
@@ -0,0 +1,384 @@
+# -*- coding: utf-8 -*-
+"""
+This module offers a parser for ISO-8601 strings
+
+It is intended to support all valid date, time and datetime formats per the
+ISO-8601 specification, with a stricter mode for the most common subset.
+"""
+from datetime import datetime, timedelta, time, date
+import calendar
+from dateutil import tz
+
+import re
+
+__all__ = ["isoparse", "Isoparser"]
+
+
+class Isoparser(object):
+    def __init__(self, sep='T', default_year=None):
+        """
+        :param sep:
+            A single character that separates date and time portions
+
+        :param default_year:
+            The default year to be used as the basis for parsing the uncommon
+            no-year date formats.
+        """
+        if (len(sep.encode('utf-8')) != 1 or
```

Comment:
You might not need to do encode at all.

You can just check the length, and if you are worried on unicode characters you are already using `ord` that will give you the unicode point.

```
>>> ord(u'语')
35821
```

Raising it as in python2 you will be calling encode in bytestrings (should not cause any issue unless using unicode characters, but quite weird)

Reply:
Ah, good call. I added the `ord` check after the `encode` portion and didn't realize it obviated the need to encode.